src/Akka.Persistence.Azure/Journal/AzureTableStorageJournal.cs

```diff
@@ -275,140 +259,117 @@ protected override void PreStart()
+                var exceptions = new List<Exception>();
+                var highSequenceNumbers = new Dictionary<string, long>();
 
                 using (var currentWrites = atomicWrites.GetEnumerator())
                 {
                     while (currentWrites.MoveNext())
                     {
                         Debug.Assert(currentWrites.Current != null, "atomicWrites.Current != null");
 
-                        var list = currentWrites.Current.Payload.AsInstanceOf<IImmutableList<IPersistentRepresentation>>();
-
-                        var batchItems = ImmutableList<ITableEntity>.Empty;
-
+                        var list = (IImmutableList<IPersistentRepresentation>) currentWrites.Current.Payload;
+                        var batchItems = new List<TableTransactionAction>();
                         foreach (var t in list)
                         {
                             var item = t;
-
                             Debug.Assert(item != null, nameof(item) + " != null");
 
-                            byte[] payloadBytes = null;
-
                             string[] tags = {};
-
                             // If the payload is a tagged payload, reset to a non-tagged payload
                             if (item.Payload is Tagged tagged)
                             {
                                 item = item.WithPayload(tagged.Payload);
-
-                                payloadBytes = _serialization.PersistentToBytes(item);
-
                                 if (tagged.Tags.Count > 0)
-                                {
                                     tags = tagged.Tags.ToArray();
-
-                                }
                             }
 
-                            if (payloadBytes == null)
-                            {
-                                payloadBytes = _serialization.PersistentToBytes(item);
-                            }
-
-                            var newItem =
-                                new PersistentJournalEntry(
-                                    item.PersistenceId,
-                                    item.SequenceNr,
-                                    payloadBytes,
-                                    item.Manifest,
-                                    tags);
+                            var payloadBytes = _serialization.PersistentToBytes(item);
+                            
+                            var newItem = new PersistentJournalEntry(
+                                persistentId: item.PersistenceId, 
+                                seqNo: item.SequenceNr,
+                                payload: payloadBytes,
+                                manifest: item.Manifest,
+                                tags: tags);
 
-                            batchItems = batchItems.Add(newItem);
+                            batchItems.Add(new TableTransactionAction(TableTransactionActionType.Add, newItem.WriteEntity()));
```

Comment:
New SDK way of building a batch data CRUD, everything is encapsulated inside a transaction-like batch